PennMobile/General/UserDefaults + Helpers.swift

```diff
@@ -389,3 +391,78 @@ extension UserDefaults {
         }
     }
 }
+
+// MARK: - Privacy Settings
+extension UserDefaults {
+    
+    // Set values for each privacy option
+    func set(_ privacyOption: PrivacyOption, to newValue: Bool) {
+        var prefs = getAllPrivacyPreferences()
+        prefs[privacyOption.rawValue] = newValue
+        saveAll(privacyPreferences: prefs)
+    }
+    
+    // Get values for each privacy option (default to false if no preference exists)
+    func getPreference(for option: PrivacyOption) -> Bool {
+        let prefs = getAllPrivacyPreferences()
+        return prefs[option.rawValue] ?? false
+    }
+    
+    // Fetch preferences from disk
+    func getAllPrivacyPreferences() -> PrivacyPreferences {
+        let decoder = JSONDecoder()
+        if let decodedData = UserDefaults.standard.data(forKey: UserDefaultsKeys.privacyPreferences.rawValue) {
+            return (try? decoder.decode(PrivacyPreferences.self, from: decodedData)) ?? .init()
+        }
+        return .init()
+    }
+    
+    // Save all privacy preferences to disk
+    func saveAll(privacyPreferences: PrivacyPreferences) {
+        let encoder = JSONEncoder()
+        if let encoded = try? encoder.encode(privacyPreferences) {
+            UserDefaults.standard.set(encoded, forKey: UserDefaultsKeys.privacyPreferences.rawValue)
+        }
+    }
+
+    private func clearPrivacyPreferences() {
+        removeObject(forKey: UserDefaultsKeys.privacyPreferences.rawValue)
+    }
+}
+
+// MARK: - Notification Settings
+extension UserDefaults {
+    // Set values for each notification option
+    func set(_ notificationOption: NotificationOption, to newValue: Bool) {
+        var prefs = getAllNotificationPreferences()
+        prefs[notificationOption.rawValue] = newValue
+        saveAll(notificationPreferences: prefs)
+    }
+    
+    // Get values for each notification option (default to false if no preference exists)
+    func getPreference(for option: NotificationOption) -> Bool {
+        let prefs = getAllNotificationPreferences()
+        return prefs[option.rawValue] ?? false
```

Comment:
I think Notification preferences should be opt-out, not opt-in, so I think the default notification preference should be true. Most apps I know are like that, and there's very little cost to the user to receive a GSR reminder. Plus, it keeps the "authorize push notification" flow simple: just yes or no. Do you agree?

Reply:
Made the default true -- great suggestion! I totally agree.